### ISSUE TITLE
SFT Qwen2.5-vl模型多机多卡训练卡住，最终导致训练失败

### ISSUE THREAD

**Issue by bang123-box** [modelscope/ms-swift#5636]
sft多机多卡训练时在加载完模型和数据集之后训练进度一致为0，一段时间后出现错误：

<img width="1280" height="691" alt="Image" src="https://github.com/user-attachments/assets/f8b8c296-b417-4ad9-88ad-a5e9da187850" />

<img width="1275" height="607" alt="Image" src="https://github.com/user-attachments/assets/1018b4cf-349e-49c3-8043-303b153461d9" />

<img width="1278" height="679" alt="Image" src="https://github.com/user-attachments/assets/aab31302-6d3e-43f4-90cb-f594ac2b12a9" />

<img width="1268" height="688" alt="Image" src="https://github.com/user-attachments/assets/9278d9f7-a589-45e5-bd90-e670f8862513" />

<img width="1277" height="575" alt="Image" src="https://github.com/user-attachments/assets/08347237-5982-46ff-b09a-43de582e7837" />

我的代码是在2台a100上实验，单机训练没问题，前两天使用2个机器训练也没问题，但是今天运行后出现上面的问题，下面是我的shell脚本
`shell

source /opt/conda/etc/profile.d/conda.sh
conda activate /nas-alinlp/zhoubb/envs/zhoubb_swift
which python

export TORCH_NCCL_TRACE_BUFFER_SIZE=2000
export NCCL_DEBUG=INFO
export NCCL_PROTO=Simple
export DS_DEBUG=1
export CUDA_DEVICE_MAX_CONNECTIONS=1

export PYTHONPATH=/nas-alinlp/zhoubb/code/demo_TabRectSet:$PYTHONPATH
export PYTHONPATH=/nas-alinlp/zhoubb/code/Table:$PYTHONPATH
export MIN_PIXELS=3136
export MAX_PIXELS=1605632

cd /nas-alinlp/zhoubb/code/Table

if [ -n "$MASTER_ADDR" ]; then
    echo "MASTER_ADDR: $MASTER_ADDR"
    echo "MASTER_PORT: $MASTER_PORT" 
    echo "WORLD_SIZE: $WORLD_SIZE"
    echo "RANK: $RANK"
else
    MASTER_ADDR=127.0.0.1
    MASTER_PORT=2$(($RANDOM % 10))$(($RANDOM % 10))15
    WORLD_SIZE=1
    RANK=0
fi

BATCH_SIZE=8
EPOCH_NUM=1
MODEL=/nas-alinlp/zhoubb/download_models/Qwen2_5-VL-3B-Instruct
GPUS_PER_NODE=8
GLOBAL_BATCH_SIZE_DIRECT=256
GRADIENT_ACCU=$((GLOBAL_BATCH_SIZE_DIRECT / $BATCH_SIZE / $GPUS_PER_NODE / $WORLD_SIZE))

echo "BATCH_SIZE: $BATCH_SIZE, GRADIENT_ACCU: $GRADIENT_ACCU, GPUS_PER_NODE: $GPUS_PER_NODE"

OUTPUT_DIR=/nas-alinlp/zhoubb/ckpt/table_sft
day=$(date +%Y%m%d)
TRAIN_FILE="acade150k_recog yewu_recog acade150k_detect yewu_detect acade50k_recog_detect yewu50k_recog_detect"
PREFIX="${day}_${TRAIN_FILE// /-}_epoch${EPOCH_NUM}-less_token_unfreezevit_qwen25vl3b"

MASTER_ADDR=$MASTER_ADDR NNODES=$WORLD_SIZE NODE_RANK=$RANK NPROC_PER_NODE=$GPUS_PER_NODE swift sft \
  --model $MODEL \
  --model_type qwen2_5_vl \
  --train_type full \
  --freeze_vit false \
  --freeze_aligner false \
  --custom_register_path plugin/own_datasets.py \
  --deepspeed zero2 \
  --torch_dtype bfloat16 \
  --num_train_epochs $EPOCH_NUM \
  --learning_rate 1e-5 \
  --per_device_train_batch_size $BATCH_SIZE \
  --eval_steps 500 \
  --save_steps 500 \
  --logging_steps 5 \
  --split_dataset_ratio 0 \
  --gradient_accumulation_steps $GRADIENT_ACCU \
  --save_total_limit 5 \
  --use_liger_kernel true \
  --attn_impl flash_attn \
  --padding_free True \
  --max_length 10800 \
  --warmup_ratio 0.05 \
  --dataloader_num_workers $BATCH_SIZE \
  --load_from_cache_file false \
  --dataset $TRAIN_FILE \
  --output_dir $OUTPUT_DIR/$PREFIX
`

我的环境如下：
absl-py                                  2.3.1
accelerate                               1.8.1
addict                                   2.4.0
aiofiles                                 24.1.0
aiohappyeyeballs                         2.6.1
aiohttp                                  3.12.13
aiosignal                                1.4.0
airportsdata                             20250706
aliyun-python-sdk-core                   2.16.0
aliyun-python-sdk-kms                    2.16.5
altair                                   5.5.0
annotated-types                          0.7.0
antlr4-python3-runtime                   4.13.2
anyio                                    4.9.0
apted                                    1.0.3
argon2-cffi                              25.1.0
argon2-cffi-bindings                     21.2.0
arrow                                    1.3.0
arxiv                                    2.2.0
astor                                    0.8.1
asttokens                                3.0.0
async-lru                                2.0.5
async-timeout                            5.0.1
attrdict                                 2.0.1
attrs                                    25.3.0
av                                       15.0.0
babel                                    2.17.0
beautifulsoup4                           4.13.4
binpacking                               1.5.2
bitsandbytes                             0.46.1
blake3                                   1.0.5
bleach                                   6.2.0
blinker                                  1.9.0
boto3                                    1.39.3
botocore                                 1.39.3
Brotli                                   1.1.0
cachetools                               5.5.2
certifi                                  2025.6.15
cffi                                     1.17.1
charset-normalizer                       3.4.2
click                                    8.2.1
cloudpickle                              3.1.1
colorama                                 0.4.6
comm                                     0.2.2
compressed-tensors                       0.9.3
contourpy                                1.3.2
cpm-kernels                              1.0.11
crcmod                                   1.7
cryptography                             45.0.5
cupy-cuda12x                             13.4.1
cycler                                   0.12.1
dacite                                   1.9.2
dashscope                                1.23.8
datasets                                 3.2.0
debugpy                                  1.8.14
decorator                                5.2.1
decord                                   0.6.0
deepspeed                                0.16.9
defusedxml                               0.7.1
Deprecated                               1.2.18
depyf                                    0.18.0
dill                                     0.3.8
diskcache                                5.6.3
Distance                                 0.1.3
distro                                   1.9.0
dnspython                                2.7.0
dotenv                                   0.9.9
duckduckgo_search                        5.3.1b1
einops                                   0.8.1
email_validator                          2.2.0
et_xmlfile                               2.0.0
evalscope                                0.16.3
evaluate                                 0.4.4
exceptiongroup                           1.3.0
executing                                2.2.0
fastapi                                  0.115.14
fastapi-cli                              0.0.7
fastjsonschema                           2.21.1
fastrlock                                0.8.3
feedparser                               6.0.11
ffmpy                                    0.6.0
filelock                                 3.18.0
fire                                     0.7.0
flash_attn                               2.7.4.post1
fonttools                                4.58.5
fqdn                                     1.5.1
frozenlist                               1.7.0
fsspec                                   2024.9.0
func_timeout                             4.3.5
future                                   1.0.0
fuzzywuzzy                               0.18.0
gguf                                     0.17.1
gitdb                                    4.0.12
GitPython                                3.1.44
google-auth                              2.40.3
google-genai                             1.24.0
googleapis-common-protos                 1.70.0
gradio                                   5.35.0
gradio_client                            1.10.4
greenlet                                 3.2.4
griffe                                   0.49.0
groovy                                   0.1.2
grpcio                                   1.73.1
h11                                      0.16.0
h2                                       4.2.0
h5py                                     3.14.0
hf-xet                                   1.1.5
hjson                                    3.1.0
hpack                                    4.1.0
httpcore                                 1.0.9
httptools                                0.6.4
httpx                                    0.28.1
huggingface-hub                          0.34.4
human-eval                               1.0.3
hyperframe                               6.1.0
idna                                     3.10
imageio                                  2.37.0
immutabledict                            4.2.1
importlib_metadata                       8.0.0
iniconfig                                2.1.0
interegular                              0.3.3
ipdb                                     0.13.13
ipykernel                                6.29.5
ipython                                  8.37.0
ipywidgets                               8.1.7
isoduration                              20.11.0
jedi                                     0.19.2
jieba                                    0.42.1
Jinja2                                   3.1.6
jiter                                    0.10.0
jmespath                                 0.10.0
joblib                                   1.5.1
json_repair                              0.47.6
json5                                    0.12.0
jsonlines                                4.0.0
jsonpointer                              3.0.0
jsonschema                               4.24.0
jsonschema-specifications                2025.4.1
jupyter                                  1.1.1
jupyter_client                           8.6.3
jupyter-console                          6.6.3
jupyter_core                             5.8.1
jupyter-events                           0.12.0
jupyter-lsp                              2.2.5
jupyter_server                           2.16.0
jupyter_server_terminals                 0.5.3
jupyterlab                               4.4.4
jupyterlab_pygments                      0.3.0
jupyterlab_server                        2.27.3
jupyterlab_widgets                       3.0.15
kiwisolver                               1.4.8
lagent                                   0.2.4
langdetect                               1.0.9
lark                                     1.2.2
latex2sympy2_extended                    1.10.2
lazy_loader                              0.4
Levenshtein                              0.27.1
liger_kernel                             0.6.0
llguidance                               0.7.30
llvmlite                                 0.44.0
lm-format-enforcer                       0.10.11
lxml                                     6.0.0
Markdown                                 3.8.2
markdown-it-py                           3.0.0
MarkupSafe                               3.0.2
math-verify                              0.8.0
matplotlib                               3.10.3
matplotlib-inline                        0.1.7
mdurl                                    0.1.2
mistral_common                           1.6.3
mistune                                  3.1.3
mmengine                                 0.10.7
mmengine-lite                            0.10.7
modelscope                               1.27.1
mpmath                                   1.3.0
ms-opencompass                           0.1.6
ms_swift                                 3.7.0.dev0     /nas-alinlp/zhoubb/code/ms-swift-main
ms-vlmeval                               0.0.18
msgpack                                  1.1.1
msgspec                                  0.19.0
multidict                                6.6.3
multiprocess                             0.70.16
narwhals                                 1.45.0
nbclient                                 0.10.2
nbconvert                                7.16.6
nbformat                                 5.10.4
nest-asyncio                             1.6.0
networkx                                 3.4.2
ninja                                    1.11.1.4
nltk                                     3.9.1
notebook                                 7.4.4
notebook_shim                            0.2.4
numba                                    0.61.2
numpy                                    1.26.4
nvidia-cublas-cu12                       12.4.5.8
nvidia-cuda-cupti-cu12                   12.4.127
nvidia-cuda-nvrtc-cu12                   12.4.127
nvidia-cuda-runtime-cu12                 12.4.127
nvidia-cudnn-cu12                        9.1.0.70
nvidia-cufft-cu12                        11.2.1.3
nvidia-cufile-cu12                       1.11.1.6
nvidia-curand-cu12                       10.3.5.147
nvidia-cusolver-cu12                     11.6.1.9
nvidia-cusparse-cu12                     12.3.1.170
nvidia-cusparselt-cu12                   0.6.2
nvidia-ml-py                             12.575.51
nvidia-nccl-cu12                         2.21.5
nvidia-nvjitlink-cu12                    12.4.127
nvidia-nvtx-cu12                         12.4.127
openai                                   1.93.0
OpenCC                                   1.1.9
opencv-contrib-python-headless           4.11.0.86
opencv-python                            4.11.0.86
opencv-python-headless                   4.11.0.86
openpyxl                                 3.1.5
opentelemetry-api                        1.26.0
opentelemetry-exporter-otlp              1.26.0
opentelemetry-exporter-otlp-proto-common 1.26.0
opentelemetry-exporter-otlp-proto-grpc   1.26.0
opentelemetry-exporter-otlp-proto-http   1.26.0
opentelemetry-proto                      1.26.0
opentelemetry-sdk                        1.26.0
opentelemetry-semantic-conventions       0.47b0
opentelemetry-semantic-conventions-ai    0.4.10
orjson                                   3.10.18
oss2                                     2.19.1
outlines                                 0.1.11
outlines_core                            0.1.26
overrides                                7.7.0
packaging                                25.0
pandas                                   2.3.0
pandocfilters                            1.5.1
parso                                    0.8.4
partial-json-parser                      0.2.1.1.post6
peft                                     0.15.2
pexpect                                  4.9.0
phx-class-registry                       4.1.0
pillow                                   11.3.0
pip                                      25.1
platformdirs                             4.3.8
playwright                               1.54.0
pluggy                                   1.6.0
portalocker                              3.2.0
prettytable                              3.16.0
prometheus_client                        0.22.1
prometheus-fastapi-instrumentator        7.1.0
prompt_toolkit                           3.0.51
propcache                                0.3.2
protobuf                                 4.25.8
psutil                                   7.0.0
ptyprocess                               0.7.0
pure_eval                                0.2.3
py-cpuinfo                               9.0.0
pyarrow                                  20.0.0
pyasn1                                   0.6.1
pyasn1_modules                           0.4.2
pycocotools                              2.0.10
pycountry                                24.6.1
pycparser                                2.22
pycryptodome                             3.23.0
pydantic                                 2.11.7
pydantic_core                            2.33.2
pydeck                                   0.9.1
pydub                                    0.25.1
pyecharts                                2.0.8
pyee                                     13.0.0
Pygments                                 2.19.2
pynvml                                   12.0.0
pyparsing                                3.2.3
pypinyin                                 0.54.0
pytest                                   8.4.1
pytest-base-url                          2.1.0
pytest-playwright                        0.7.0
python-dateutil                          2.9.0.post0
python-dotenv                            1.1.1
python-json-logger                       3.3.0
python-Levenshtein                       0.27.1
python-multipart                         0.0.20
python-slugify                           8.0.4
pytz                                     2025.2
PyYAML                                   6.0.2
pyzmq                                    27.0.0
qwen-vl-utils                            0.0.11
rank-bm25                                0.2.2
RapidFuzz                                3.13.0
ray                                      2.47.1
referencing                              0.36.2
regex                                    2024.11.6
requests                                 2.32.4
rfc3339-validator                        0.1.4
rfc3986-validator                        0.1.1
rich                                     14.0.0
rich-toolkit                             0.14.8
rouge                                    1.0.1
rouge-chinese                            1.0.3
rouge_score                              0.1.2
rpds-py                                  0.26.0
rsa                                      4.9.1
ruff                                     0.12.2
s3transfer                               0.13.0
sacrebleu                                2.5.1
safehttpx                                0.1.6
safetensors                              0.5.3
scikit-image                             0.25.2
scikit-learn                             1.7.0
scipy                                    1.15.3
seaborn                                  0.13.2
semantic-version                         2.10.0
Send2Trash                               1.8.3
sentence-transformers                    5.0.0
sentencepiece                            0.2.0
setuptools                               78.1.1
sgmllib3k                                1.0.0
shellingham                              1.5.4
simplejson                               3.20.1
six                                      1.17.0
smmap                                    5.0.2
sniffio                                  1.3.1
socksio                                  1.0.0
sortedcontainers                         2.4.0
soupsieve                                2.7
stack-data                               0.6.3
starlette                                0.46.2
streamlit                                1.46.1
sty                                      1.0.6
swankit                                  0.2.4
swanlab                                  0.6.4
sympy                                    1.13.1
tabulate                                 0.9.0
tenacity                                 8.5.0
tensorboard                              2.19.0
tensorboard-data-server                  0.7.2
termcolor                                3.1.0
terminado                                0.18.1
text-unidecode                           1.3
threadpoolctl                            3.6.0
tifffile                                 2025.5.10
tiktoken                                 0.9.0
timeout-decorator                        0.5.0
timm                                     1.0.16
tinycss2                                 1.4.0
tokenizers                               0.21.4
toml                                     0.10.2
tomli                                    2.2.1
tomlkit                                  0.13.3
torch                                    2.6.0
torchaudio                               2.6.0
torchvision                              0.21.0
tornado                                  6.5.1
tqdm                                     4.67.1
traitlets                                5.14.3
transformers                             4.51.3
transformers-stream-generator            0.0.5
triton                                   3.2.0
trl                                      0.19.0
typer                                    0.16.0
types-python-dateutil                    2.9.0.20250516
typing_extensions                        4.14.0
typing-inspection                        0.4.1
tzdata                                   2025.2
uri-template                             1.3.0
urllib3                                  2.5.0
uvicorn                                  0.35.0
uvloop                                   0.21.0
validators                               0.35.0
vllm                                     0.8.5.post1
watchdog                                 6.0.0
watchfiles                               1.1.0
wcwidth                                  0.2.13
webcolors                                24.11.1
webencodings                             0.5.1
websocket-client                         1.8.0
websockets                               15.0.1
Werkzeug                                 3.1.3
wheel                                    0.45.1
widgetsnbextension                       4.0.14
word2number                              1.1
wrapt                                    1.17.2
xformers                                 0.0.29.post2
xgrammar                                 0.1.18
xlsxwriter                               3.2.5
xtuner                                   0.1.23
xxhash                                   3.5.0
yapf                                     0.43.0
yarl                                     1.20.1
zipp                                     3.23.0
zstandard                                0.23.0

**Comment by Jintao-Huang** [comment]
试试3.7.3的版本

**Comment by bang123-box** [comment]
好的，我更新到3.7.3重新再试一下

**Comment by bang123-box** [comment]
升级到3.7.3可以训练了，谢谢大佬
